### DISCUSSION TITLE
Previous/Next sibling actions should select nearest code block when code blocks

### DISCUSSION THREAD

**Discussion by aral** [helix-editor/helix#10358] [General]
Given:

```js
1 const hello = 'hello'
2
3 console.log(hello)
```

If I’m on 1 and hit next sibling (<kbd>Alt</kbd><kbd>→</kbd>), I get to 3.
If I’m on 2 and hit next sibling, I get nothing.
If I’m on 3 and hit previous sibling  (<kbd>Alt</kbd><kbd>←</kbd>), I get 1.
If I’m on 2 and hit previous sibling, I get nothing.

I understand why this is (because I’m between code blocks) but I feel it would help if this included special case logic to work as you would assume it does.

**Comment by the-mikedavis** [comment]
Actually in this case next and prev sibling do nothing because line 2 is on the root of the syntax tree (the `program` node which is the entire document) and the root has no siblings. You can check the subtree the primary cursor is on with `:tree-sitter-subtree`. It might be possible to descend into child nodes in the case where we recognize we're on the root but I think it would be confusing behavior since the next/prev-sibling commands generally climb the tree when no sibling nodes are detected.
